plugin/autosource.vim

```diff
@@ -76,64 +76,44 @@ function! s:CheckHash(path)
     return 1
 endfunction
 
-" Source all `.vimrc` files in your pwd and parents up to your home dir
-function! SourceParentRCs(path, uuid)
-    " Prevent this function from being invoked by multiple autocmds
-    if exists('g:autosource_running') && g:autosource_running !=# a:uuid
-        return
-    elseif has('macunix')
-        if !executable('uuid')
-            echo 'Please install the `uuid` command'
-        endif
-        let g:autosource_running = system('uuid')
-    elseif has('unix')
-        if !executable('uuidgen')
-            echo 'Please install the `uuidgen` command'
-        endif
-        let g:autosource_running = system('uuidgen')
-    endif
-
-    if a:path ==# ''
-        let cur = expand('%:p:h')
-    else
-        let cur = a:path
-    endif
-
-    " don't source outside of home dir
-    if cur !~ $HOME
-        return
-    endif
+let s:fnames = ['.vimrc', '.vimrc.lua']
```

Comment:
👍 

A doc update to indicate that this will cause known vim confs to be unrecognized would be nice since the lookup is done via filename hash. Let me know if you feel like adding that, otherwise I can just merge and add it afterwards.